File: android/src/main/java/com/mattermost/networkclient/enums/APIClientEvents.kt
package com.mattermost.networkclient.enums

enum class ApiClientEvents(val event: String) {
    DOWNLOAD_PROGRESS("ApiClient-DownloadProgress"),
    UPLOAD_PROGRESS("ApiClient-UploadProgress"),
    CLIENT_ERROR("ApiClient-Error"),
}

enum class SslErrors(val event: Int) {
    SERVER_TRUST_EVALUATION_FAILED(-298)
}

File: ios/ApiClient/ApiClientEvents.swift
import Foundation

@objc public protocol ApiClientDelegate {
    func sendEvent(name: String, result: Dictionary<String, Any>?)
}

enum ApiEvents: String, CaseIterable {
    case UPLOAD_PROGRESS = "ApiClient-UploadProgress"
    case DOWNLOAD_PROGRESS = "ApiClient-DownloadProgress"
    case CLIENT_ERROR = "ApiClient-Error"
}

extension ApiClientWrapper {
    @objc
    public static var supportedEvents: [String] {
        return ApiEvents.allCases.map(\.rawValue)
    }
}

